src/transactionsFactories/accountTransactionsFactory.ts

```diff
@@ -0,0 +1,120 @@
+    gasLimitStorePerByte: bigint;
+    gasLimitSetGuardian: bigint;
+    gasLimitGuardAccount: bigint;
+    gasLimitUnguardAccount: bigint;
+    extraGasLimitForGuardedTransaction: bigint;
+}
+
+export class AccountTransactionsFactory {
+    private readonly config: IConfig;
+
+    constructor(options: { config: IConfig }) {
+        this.config = options.config;
+    }
+
+    createTransactionForSavingKeyValue(options: {
+        sender: IAddress;
+        keyValuePairs: Map<Uint8Array, Uint8Array>;
+    }): Transaction {
+        const functionName = "SaveKeyValue";
+        const keyValueParts = this.computeDataPartsForSavingKeyValue(options.keyValuePairs);
+        const dataParts = [functionName, ...keyValueParts];
+        const extraGas = this.computeExtraGasForSavingKeyValue(options.keyValuePairs);
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: extraGas,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForSettingGuardian(options: {
+        sender: IAddress;
+        guardianAddress: IAddress;
+        serviceID: string;
+    }): Transaction {
+        const dataParts = [
+            "SetGuardian",
+            Address.fromBech32(options.guardianAddress.bech32()).toHex(),
+            Buffer.from(options.serviceID).toString("hex"),
+        ];
+
+        const gasLimit = this.config.gasLimitSetGuardian + this.config.extraGasLimitForGuardedTransaction;
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForGuardingAccount(options: { sender: IAddress }): Transaction {
+        const dataParts = ["GuardAccount"];
+        const gasLimit = this.config.gasLimitGuardAccount + this.config.extraGasLimitForGuardedTransaction;
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForUnguardingAccount(options: { sender: IAddress }): Transaction {
+        const dataParts = ["UnGuardAccount"];
+        const gasLimit = this.config.gasLimitUnguardAccount + this.config.extraGasLimitForGuardedTransaction;
+
+        const transaction = new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: gasLimit,
+            addDataMovementGas: true,
+        }).build();
+        transaction.options = 2;
```

Comment:
I think it makes sense to leave it up to the higher level applications to set all the necessary stuff. Since we also don't add the gas for guarded transactions we can also omit this.